testing/test_capture.py

```diff
@@ -668,6 +669,46 @@ def test_a(fix):
         reprec = testdir.inline_run()
         reprec.assertoutcome(passed=1)
 
+    def test_keeps_sys_fd_encoding(self, testdir, monkeypatch):
+        import subprocess
+
+        # Get the real encoding, without any outer pytest.
+        # NOTE: py27: "None" with shell=False, "" with shell=True!?
+        p = subprocess.Popen(
+            [sys.executable, "-c", "import sys; print(sys.stdout.encoding)"],
```

Comment:
So since sys.stdout is a pipe and therefore has no encoding, how about

```suggestion
            [sys.executable, "-c", "import sys; print(sys.stderr.encoding)"],
```
?

Reply:
Alternatively, how about checking `sys.__stdout__.encoding`?  That should be pointing to the real stdout.

Reply:
In fact the entire test could probably be replaced with

```python
    assert sys.stdint.encoding == sys.__stdin__.encoding
    assert sys.stdout.encoding == sys.__stdout__.encoding
    assert sys.stderr.encoding == sys.__stderr__.encoding
```

Reply:
Scratch that idea, pytest overwrites `sys.__stdout__` and friends.  :-(

(I think that is rude and pytest shouldn't do that, but I assume you have your reasons.)

Reply:
@mgedmin we shouldnt do quite as much, but fd based capture is rather tricky in some sense

i hope in future we can limit the pure python parts to more pure pure python approaches - and leave the file descriptors alone (its a big problematic mess)

Reply:
1. Re using `sys.stderr.encoding`: this is also 'None\n' with py27 (on Linux), or "" with (shell=True).
This is related to using not `-s` with the outer pytest after all, and might indicate a still existing problem with this PR - i.e. it should not be None also with py27 here in the first place maybe?!

2. Good point re "sys.__stdout__" etc, I've created https://github.com/pytest-dev/pytest/pull/4407 for this.. seems to be relevant/done for py36+/windows though only.

I've simplified the test according to your suggestion, thanks!